src/modules/share/share.repository.ts

```
@@ -0,0 +1,257 @@
+  ): Promise<Share>;
+  findByFolderIdAndUser(
+    folderId: FolderAttributes['id'],
+    userId: UserAttributes['id'],
+  ): Promise<Share>;
+}
+
+@Injectable()
+export class SequelizeShareRepository implements ShareRepository {
+  constructor(
+    @InjectModel(ShareModel)
+    private shareModel: typeof ShareModel,
+    @InjectModel(FileModel)
+    private fileModel: typeof FileModel,
+    @InjectModel(FolderModel)
+    private folderModel: typeof FolderModel,
+    @InjectModel(UserModel)
+    private userModel: typeof UserModel,
+  ) {}
+
+  async findById(id: ShareAttributes['id']) {
+    const share = await this.shareModel.findByPk(id, {
+      include: [this.fileModel, this.userModel, this.folderModel],
+    });
+    return share ? this.toDomain(share) : null;
+  }
+
+  async findByFileIdAndUser(
+    fileId: FileAttributes['id'],
+    userId: UserAttributes['id'],
+  ): Promise<Share> {
+    const share = await this.shareModel.findOne({
+      where: { fileId, userId },
+      include: [this.fileModel, this.userModel],
+    });
+    return share ? this.toDomain(share) : null;
+  }
+
+  async findByFolderIdAndUser(
+    folderId: FolderAttributes['id'],
+    userId: UserAttributes['id'],
+  ) {
+    const share = await this.shareModel.findOne({
+      where: { folderId, userId },
+      include: [this.folderModel, this.userModel],
+    });
+    return share ? this.toDomain(share) : null;
+  }
+
+  async findByToken(token: string): Promise<Share | null> {
+    const share = await this.shareModel.findOne({
+      where: { token },
+      include: [this.fileModel, this.folderModel, this.userModel],
+    });
+    if (!share) {
+      throw new NotFoundException('share not found');
+    }
+    return this.toDomain(share);
+  }
+  async create(share: Share): Promise<void> {
+    const shareModel = this.toModel(share);
+    delete shareModel.id;
+    await this.shareModel.create(shareModel);
+  }
+
+  async update(share: Share): Promise<true> {
```

If the return type is always true to emphasize that the operation has ended properly, but an error is thrown if something fails (instead of returning a false), this loses consistency. It is clearer to just assume that the update has been successful if any error happens, to keep the consistency here.